openeo/rest/connection.py

```diff
@@ -228,6 +228,9 @@ def __init__(
         self._auth_config = auth_config
         self._refresh_token_store = refresh_token_store
 
+    def _repr_html_(self):
```

Comment:
As a user If find this `Connection._repr_html_` pretty intrusive.

I normally start my notebooks with a cell like this:

```
con = openeo.connect("https://openeo.vito.be")
con.authenticate_oidc()
```

Normally I just get a simple output cell with `<Connection to 'https://openeo.vito.be/openeo/1.0/' with BearerAuth>`, which gives me the essential information: what backend URL am I using and am I logged in?

But with this PR I suddenly get a large rich HTML element:
- it only shows the root url, but not the backend URL after version discovery
- it doesn't show if the connection is authenticated
-  I have to scroll almost a full window height to get to the next cell

Reply:
I wasn't aware that the authentication processes return a Connection object. It was meant that only printing the connection object itself gives this overview. Is there a way to make this happen?

I'm personally found the `<Connection to 'https://openeo.vito.be/openeo/1.0/' with BearerAuth>` always pretty strange. It gives a URL that I did not specify with some strange Auth thing that I (as a normal user) don't know anything about. I would expect something like `<Connection established to https://openeo.vito.be, version {x.x}, authenticated via {IP}>`.

Reply:
Moved to #216